pkg/storage/secret/metadata/secret_mgmt.go

```diff
@@ -16,15 +16,15 @@ func (s *secureValueMetadataStorage) storeInKeeper(ctx context.Context, sv *secr
 		return "", fmt.Errorf("store by ref in keeper")
 	}
 
-	keeperType, keeperConfig, err := s.keeperMetadataStorage.GetKeeperConfig(ctx, sv.Namespace, sv.Spec.Keeper)
+	keeperConfig, err := s.keeperMetadataStorage.GetKeeperConfig(ctx, sv.Namespace, sv.Spec.Keeper)
 	if err != nil {
 		return "", fmt.Errorf("get keeper config: %w", err)
 	}
 
 	// Store in keeper.
-	keeper, ok := s.keepers[keeperType]
-	if !ok {
-		return "", fmt.Errorf("could not find keeper: %s", keeperType)
+	keeper, err := s.keeperService.KeeperForConfig(keeperConfig)
```

Comment:
On OSS, this method is probably inlined because it just returns the field from the struct, and in Enterprise we do a small check to see if the config is `nil`, if it is, we call into the OSS impl, otherwise it's a map lookup for 3rdparty keepers.